File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n < 1 || n > 64)
        {
            throw new ArgumentOutOfRangeException("square out of range");
        }
        return (ulong)Math.Pow(2, n-1);
    }

    public static ulong Total()
    {
        ulong totalGrains = 0;
        for (int i = 1; i <= 64; i++)
        {
            totalGrains += Square(i);
        }
        return totalGrains;
    }
}

File: go/grains/grains.go
package grains

import (
	"errors"
)

const totalGrains = 1<<64 - 1

// Square returns 2 raised to the power of the input minus 1
// Returns an error if the input is < 1 or > 64
func Square(input int) (uint64, error) {
	if input < 1 || input > 64 {
		return 0, errors.New("Input must be between 1 and 64")
	}
	return 1 << (input - 1), nil
}

// Total returns the max value of a uint64
func Total() uint64 {
	return totalGrains
}
